mypy/test/data.py

```diff
@@ -583,16 +588,19 @@ def split_test_cases(parent: 'DataSuiteCollector', suite: 'DataSuite',
     cases = re.split(r'^\[case ([a-zA-Z_0-9]+)'
                      r'(-writescache)?'
                      r'(-only_when_cache|-only_when_nocache)?'
+                     r'(-posix|-windows)?'
                      r'(-skip)?'
                      r'\][ \t]*$\n', data,
```

Comment:
While you're here, can you move `data` to a line by itself? It's hiding here behind the end of the regex.